src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/PiglinListener.java

```diff
@@ -1,25 +1,34 @@
 package io.github.thebusybiscuit.slimefun4.implementation.listeners;
 
+import java.util.Set;
+import java.util.concurrent.ThreadLocalRandom;
+
 import org.bukkit.Material;
 import org.bukkit.entity.EntityType;
+import org.bukkit.entity.Item;
 import org.bukkit.entity.Piglin;
 import org.bukkit.entity.Player;
 import org.bukkit.event.EventHandler;
 import org.bukkit.event.Listener;
+import org.bukkit.event.entity.EntityDropItemEvent;
 import org.bukkit.event.entity.EntityPickupItemEvent;
 import org.bukkit.event.player.PlayerInteractEntityEvent;
 import org.bukkit.inventory.EquipmentSlot;
 import org.bukkit.inventory.ItemStack;
 
+import io.github.thebusybiscuit.slimefun4.core.attributes.PiglinBarterDrop;
+import io.github.thebusybiscuit.slimefun4.core.attributes.RandomMobDrop;
 import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
 import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
 
 /**
- * This {@link Listener} prevents a {@link Piglin} from bartering with a {@link SlimefunItem}.
+ * This {@link Listener} prevents a {@link Piglin} from bartering with a {@link SlimefunItem} as well as
+ * listens to the EntityDropItemEvent to inject a {@link PiglinBarterDrop} if a dropChance() check passes.
  *
- * @author poma123
+ * @author poma123, dNiym
```

Comment:
Make a new author tag below this one.
You can have multiple, no need to use a comma here